cvat-ui/src/components/attribute-editor/attribute-editor.tsx

```diff
@@ -0,0 +1,358 @@
+        filteredShapes = filteredShapes.filter(identityFieldCondition(frame, identityAttrId, identityId));
+        if (filteredShapes.length > 1) {
+            dangerousFrames.push(frame);
+        } else {
+            for (let shape of filteredShapes) {
+                affectedShapes.push([frame, shape]);
+            }
+        }
+    }
+
+    if (dangerousFrames.length > 0) {
+        return `Cannot change the attribute as it affects multiple objects on frames ${dangerousFrames.join(', ')}.`;
+    }
+    if (affectedShapes.length == 0) {
+        return `There are no affected shapes within the selected range.`;
+    }
+    let attrInfo = labelsInfo.attrInfo(attributeName);
+    const doUpdate = (batch) => {
+        batch.forEach(([frame, shape, value]) => {
+            if (typeof (value) === 'undefined') {
+                delete shape._attributes.mutable[frame][attributeName];
+                this.notify('attributes');
+                return;
+            }
+            const attrValue = LabelsInfo.normalize(attrInfo.type, value);
+            if (attrInfo.mutable) {
+                shape._attributes.mutable[frame] = shape._attributes.mutable[frame] || {};
+                shape._attributes.mutable[frame][attributeName] = attrValue;
+                shape._setupKeyFrames();
+            } else {
+                shape._attributes.immutable[attributeName] = attrValue;
+            }
+            shape.notify('attributes');
+        });
+    };
+    const forwardData = affectedShapes.map(([frame, shape]) => [frame, shape, attributeValue]);
+    const backwardData = affectedShapes.map(([frame, shape]) => {
+        let oldValue;
+        if (attrInfo.mutable) {
+            oldValue = shape._attributes.mutable[frame] ? shape._attributes.mutable[frame][attributeName] : undefined;
+        } else {
+            oldValue = shape._attributes.immutable[attributeName];
+        }
+        return [frame, shape, oldValue];
+    });
+
+    doUpdate(forwardData);
+    window.cvat.addAction('Bulk Update Attribute', () => {
+        doUpdate(backwardData);
+    }, () => {
+        doUpdate(forwardData);
+    }, start);
+    return null;
+}
+
+function AttributeEditor({ start: minFrame, stop: maxFrame, labelsInfo, shapeCollection }) {
+    const [visible, setVisible] = React.useState(false);
+    const [start, setStart] = React.useState(minFrame);
+    const [stop, setStop] = React.useState(maxFrame);
+    const [labelId, setLabelId] = React.useState(null);
+    const [identityId, setIdentityId] = React.useState(null);
+    const [identityAttrId, identityAttrName] = getIdentityAttribute(labelId, labelsInfo);
+    const [attributeName, setAttributeName] = React.useState(null);
+    const [attributeValue, setAttributeValue] = React.useState(null);
+    React.useLayoutEffect(() => {
+        if (identityAttrId == null && !(identityId == null)) {
+            setIdentityId(null);
+        }
+    });
+    const buttonStyle = {
```

Comment:
nit: ideally you should set class and put style in CSS

Reply:
It's definitely better to put styles in CSS, but this component doesn't have much of them so I was lazy to do so.